sys/dev/athn/if_athn_usb.c

```diff
@@ -603,16 +602,102 @@ athn_usb_verify_fw_ver(struct athn_softc *sc, struct ar_wmi_fw_version *img_ver)
 	return error;
 }
 
+static void
+athn_set_channel(struct ieee80211com *ic)
+{
+	struct athn_softc *sc = ic->ic_softc;
+
+	ATHN_LOCK(sc);
+	(void) athn_usb_switch_chan(sc, ic->ic_curchan, NULL);
+	ATHN_UNLOCK(sc);
+}
+
+static struct athn_usb_data *
+_athn_getbuf(struct athn_softc *sc)
+{
+	struct athn_usb_data *bf;
```

Comment:
I don't think we should initialize them here as NULL, we are in kernel space and this will impact negatively speed of our driver, although only a small amount. It would be better to change order of variable creation so we can do something like this:

struct athn_usb_softc *usc = sc->sc_usc;
struct athn_usb_data *bf = STAILQ_FIRST(&usc->sc_tx_inactive);

Although I think this should be okay to leave as it is, we immediately assign value to `bf` in line ~621 and this function is small so it shouldn't be an issue where someone in future will do something unintentional here. Also this function is almost identical to already existing code from otus driver which further suggest that this is rather okay.